### ISSUE TITLE
There's an error when delete a folder under watched

### ISSUE THREAD

**Issue by richard-chen-1985** [gulpjs/gulp#945]
My code like this:

```
gulp.watch('src/**', function(event) {
    //......
})
```
If i delete a folder in `src/`, an error throwed like:

```
events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: watch EPERM
    at errnoException (fs.js:1030:11)
    at FSEvent.FSWatcher._handle.onchange (fs.js:1048:26)
```

**Comment by MystK** [comment]
I have the same issue, and it is an issue with gaze upon further testing.

Here's my example code with gaze which outputted the same error.

```javascript
var Gaze = require('gaze').Gaze;
var gaze = new Gaze('backend/**');
gaze.on('all', function(event, filepath) {console.log(event)});
```
outputted
```
events.js:141
      throw er; // Unhandled 'error' event
            ^
Error: watch EPERM
    at exports._errnoException (util.js:734:11)
    at FSEvent.FSWatcher._handle.onchange (fs.js:1203:26)
```
I'm not sure if there is an issue open on gaze, but this will probably get merged with [#651](https://github.com/gulpjs/gulp/issues/651)

**Comment by richard-chen-1985** [comment]
thanks
